server/routes/charts.js

```diff
@@ -143,29 +143,22 @@ router
   })
   .get(ensureAuthenticated, async (req, res) => {
     try {
+      const chartList = []
       const { dataDb, appDb } = req.app.locals
-      const charts = await dataDb
-        .collection(collections.charts)
-        .find({
-          $or: [
-            { owner: req.user },
-            { sharedWith: req.user },
-            { public: true },
-          ],
-        })
-        .toArray()
-      const chartList = await Promise.all(
-        charts.map(async (chart) => {
-          const { owner } = chart
-          if (owner !== req.user) {
-            const chartOwner = await appDb
-              .collection(collections.users)
-              .findOne({ uid: owner }, { projection: { _id: 0, icon: 1 } })
-            chart.icon = chartOwner.icon
-            return chart
-          } else return chart
-        })
-      )
+      const chartListCursor = await dataDb.collection(collections.charts).find({
+        $or: [{ owner: req.user }, { sharedWith: req.user }, { public: true }],
+      })
+      while (await chartListCursor.hasNext()) {
+        const chart = await chartListCursor.next()
+        chart.chartOwner = await appDb
+          .collection(collections.users)
+          .findOne(
+            { uid: chart.owner },
+            { projection: { _id: 0, icon: 1, uid: 1, name: '$display_name' } }
+          )
+        chartList.push(chart)
+      }
```

Comment:
is this a more performant way to add the chart owner than what we had previously? Curious why we're changing to use the cursor instead of the array.

Reply:
yes, it is more performant, we don't create multiple copies of the data to memory. The cursor is asynchronous so we don't have to do things like Promise.all and an async map.